### PR TITLE
ipc: buffer: Fix incorrect memory management in ipc_buffer_new()

### PR DESCRIPTION
**Pull request by akloniex**
Signed-off-by: Artur Kloniecki <arturx.kloniecki@linux.intel.com>

### CODE REVIEW

File: src/ipc/ipc.c [hunk 1]
```diff
@@ -267,7 +267,7 @@ int ipc_buffer_new(struct ipc *ipc, struct sof_ipc_buffer *desc)
 	buffer = buffer_new(desc);
 	if (buffer == NULL) {
 		trace_ipc_error("ipc_buffer_new() error: buffer_new() failed");
-		rfree(ibd);
```
Comment (akloniex) [inline src/ipc/ipc.c]: At this point ibd is NULL, because of the if statement few lines above.

Comment (ranj063) [reply src/ipc/ipc.c]: @akloniex there's a similar one on line 276 as well?

Comment (akloniex) [reply src/ipc/ipc.c]: @ranj063 around line 276, we should call buffer_free(), instead rfree(buffer), that's a mem leak.
If allocation of ibd fails, then _properly_ free the buffer that was allocated with buffer_new at line 267, and abandon whole operation.
I suppose that I could add that as separate patch to this PR.

**Comment by lgirdwood** [comment]
CI known issue.

**Pull request merged**